File: Examples/Communication/CSharp/Src/Server/Services/Mail.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Communication;

namespace Server.Services
{
    public partial class ServerImplementation : IApplicationServer
    {
        public GetUserProfileReply GetUserProfile(GetUserProfileRequest r)
        {
            //TODO

            return GetUserProfileReply.CreateNotExist();
        }
    }
}

File: Examples/Communication/CPP/Src/Server/Services/Mail.cpp
#include "Services/ServerImplementation.h"

#include <memory>
#include <string>

using namespace Communication;
using namespace Server::Services;

/// <summary>获取用户信息</summary>
std::shared_ptr<GetUserProfileReply> ServerImplementation::GetUserProfile(std::shared_ptr<GetUserProfileRequest> r)
{
    return GetUserProfileReply::CreateNotExist();
}
